File: microservices-app/sentiment-analysis/infra/api/endpoints/commands/sentiment.py
from bottle import request

from infra.api.endpoints.response import json_response
from domain.useCases.sentimentClassifier import classifyText

class SentimentEndpointCommand(object):
  def register(self, app):
    app.route('/sentiment', 'POST', self.execute_post_query)

  def execute_post_query(self):
    post_content = request.json
    text_to_analyse = post_content['text']

    output = {
      'analysedText': text_to_analyse,
      'sentiment': classifyText(text_to_analyse)
    }

    return json_response(output)

File: microservices-app/tweets/Infra/clients/sentiment/sentiment.js
const dependencies = {
  config: require('Infra/config/sentiment'),
  axios: require('axios')
}

const classifyText = async (text, injection) => {
  const { axios, config } = Object.assign({}, dependencies, injection)

  const postData = { text }

  const response = await axios.post(config.url, postData)

  return response.data
}

module.exports = {
  classifyText
}
